src/SourceBuild/nuget-client/0001-support-rid-agnostic-publishing.patch

```diff
@@ -0,0 +1,31 @@
+From 31a4ea0785c8e98a2808ebf55928f67bf8f43079 Mon Sep 17 00:00:00 2001
+From: Jeremy Koritzinsky <jkoritzinsky@gmail.com>
+Date: Fri, 7 Mar 2025 11:22:09 -0800
+Subject: [PATCH] Don't publish RID-agnostic nuget packages when we only want
+ RID-specific artifacts
+
+This is necessary to enable us to deduplicate assets in the VMR.
+
+Backport: https://github.com/NuGet/NuGet.Client/pull/6306
+---
+ eng/Publishing.props | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/eng/Publishing.props b/eng/Publishing.props
+index 5fe1ceb1a79..664c36c89c4 100644
+--- a/eng/Publishing.props
++++ b/eng/Publishing.props
+@@ -2,7 +2,12 @@
+ 
+   <ItemGroup>
+     <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
+-              PublishFlatContainer="false" />
++              PublishFlatContainer="false"
++              Condition="'$(EnableDefaultRidSpecificArtifacts)' != 'true'" />
++   <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
++             PublishFlatContainer="false"
++             Visibility="Vertical"
++             Condition="'$(EnableDefaultRidSpecificArtifacts)' == 'true' and '$(DotNetBuildOrchestrator)' == 'true'" />
+  </ItemGroup>
```

Comment:
I'm not sure how to read this. In both cases the glob expression is the same. Is the difference the Visbility metadata? Why do we need the `EnableDefaultRidSpecificArtifacts` condition given that nuget-client doesn't produce RID specific packages?

Reply:
Yes the difference is the visibility metadata.

All of this logic is replicating the built-in behavior from Arcade with NuGet.Client's package layout (and the assumption that none of those packages are rid-specific).

Reply:
OK. Can you please add a comment (doesn't need to be in this patch) to the PR so that others looking at it will understand those lines as well? I think this is quite complex to follow. Nit, you could simplify this block to:

```xml
     <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
              PublishFlatContainer="false">
      <Visibility Condition="'$(EnableDefaultRidSpecificArtifacts)' == 'true' and '$(DotNetBuildOrchestrator)' == 'true'">Vertical</Visibility>
    </Artifact>
```